src/examples/token-manager.new.ts

```diff
@@ -0,0 +1,380 @@
+    );
+
+    ftProof.verifyIf(vk, Bool(false));
+
+    const token = new FungibleToken(this.tokenAddress.getAndRequireEquals());
+    const sender = this.sender.getUnconstrained();
+
+    await token.transferCustom(
+      this.address,
+      sender,
+      amount,
+      ftProof,
+      vk,
+      vKeyMap
+    );
+  }
+}
+
+const proofsEnabled = true;
+
+const localChain = await Mina.LocalBlockchain({
+  proofsEnabled,
+  enforceTransactionLimits: false,
+});
+Mina.setActiveInstance(localChain);
+const fee = 1e8;
+
+const [deployer, alexa] = localChain.testAccounts;
+const fungibleTokenKeypair = PrivateKey.randomKeypair();
+const tokenManagerKeypair = PrivateKey.randomKeypair();
+const fungibleTokenAddress = fungibleTokenKeypair.publicKey;
+const tokenManagerAddress = tokenManagerKeypair.publicKey;
+
+const vKeyMap = new VKeyMerkleMap();
+const dummyVKey = await VerificationKey.dummy();
+
+console.log('---- Addresses ----');
+console.log('Deployer:', deployer.toBase58());
+console.log('Alexa:', alexa.toBase58());
+console.log('FungibleToken Contract:', fungibleTokenAddress.toBase58());
+console.log('TokenManager Contract:', tokenManagerAddress.toBase58());
+
+console.log('\n---- Compiling Contracts & ZkPrograms ----');
+await EcdsaAndKeccakProgram.compile();
+await FungibleToken.compile();
+await TokenManager.compile();
+console.log('Compilation complete.');
+
+console.log('\n---- Deploying FungibleToken Contract ----');
+const fungibleToken = new FungibleToken(fungibleTokenAddress);
+
+const deployFtTx = await Mina.transaction(
+  { sender: deployer, fee },
+  async () => {
+    AccountUpdate.fundNewAccount(deployer, 2);
+    await fungibleToken.deploy({
+      symbol: 'TKN',
+      src: 'https://github.com/o1-labs-XT/fungible-token-standard/blob/main/src/NewTokenStandard.ts',
+    });
+    await fungibleToken.initialize(
+      tokenManagerAddress,
+      UInt8.from(9),
+      MintConfig.default,
+      new MintParams({
+        fixedAmount: UInt64.from(100),
+        minAmount: UInt64.from(20),
+        maxAmount: UInt64.MAXINT(),
+      }),
+      BurnConfig.default,
+      new BurnParams({
+        fixedAmount: UInt64.from(100),
+        minAmount: UInt64.from(20),
+        maxAmount: UInt64.MAXINT(),
+      }),
+      MintDynamicProofConfig.default,
+      BurnDynamicProofConfig.default,
+      TransferDynamicProofConfig.default,
+      UpdatesDynamicProofConfig.default
+    );
+  }
+);
+deployFtTx.sign([deployer.key, fungibleTokenKeypair.privateKey]);
+await deployFtTx.prove();
+await deployFtTx.send();
+console.log(
+  'FungibleToken deployed and initialized with TokenManager as admin.'
+);
+
+const dummyFtProof = await generateDummyDynamicProof(
+  fungibleToken.deriveTokenId(),
+  deployer
+);
+console.log('Dummy FT SideloadedProof generated.');
+
+console.log('\n---- Deploying TokenManager Contract ----');
+const tokenManager = new TokenManager(tokenManagerAddress);
+const deployTmTx = await Mina.transaction(
```

Comment:
Is the deploy order important?  Might be worth mentioning if it is.  It feels weird that both contracts need to reference each other by public key, but they aren't created together.  Could we even put them in one atomic transaction, or is that too many AUs for one tx?

Reply:
I haven't tried combining them into a single transaction.

The reason we reference the Token Manager contract when deploying the Fungible Token is because we want the Token Manager to be the token owner. Conversely, we reference the Fungible Token address when deploying the Token Manager because we'll be performing transfers using that token.

If you're okay with it, we can keep them as separate transactions and I can add comments in the code to clarify this why they're referencing each other?

Reply:
I'm fine with it as is.  If this were a production deployment, it would be better as a single atomic transaction.  You could conceivably deploy the first thing successfully, then fail to deploy the second thing, so the first thing will be useless.  It would also just be less code overall, and fewer "setup steps".